sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/DigitalTwinsRelationshipAsyncTest.java

```diff
@@ -268,39 +269,77 @@ public void relationshipListOperationWithMultiplePages(HttpClient httpClient, Di
 
             // Connect the created twins via relationships
             String floorContainsRoomPayload = getRelationshipWithPropertyPayload(roomTwinId, CONTAINS_RELATIONSHIP, "isAccessRestricted", true);
+            String roomContainedInFloorPayload = getRelationshipPayload(floorTwinId, CONTAINED_IN_RELATIONSHIP);
 
             // Create large number of relationships to test paging functionality
             // Relationship list api does not have max item count request option so we have to create a large number of them to trigger paging functionality from the service.
             // Create relationships from Floor -> Room
-            final CountDownLatch createRelationshipsLatch = new CountDownLatch(BULK_RELATIONSHIP_COUNT);
-
-            for (int i = 0 ; i< BULK_RELATIONSHIP_COUNT ; i++) {
+            for (int i = 0; i < pageSize++; i++) {
                 String relationshipId = FLOOR_CONTAINS_ROOM_RELATIONSHIP_ID + this.testResourceNamer.randomUuid();
-                asyncClient.createRelationship(floorTwinId, relationshipId, deserializeJsonString(floorContainsRoomPayload, BasicRelationship.class), BasicRelationship.class)
-                    .doOnSuccess(s -> createdRelationshipIds.add(relationshipId))
-                    .doOnTerminate(createRelationshipsLatch::countDown)
-                    .subscribe();
+                StepVerifier.create(
+                    asyncClient.createRelationship(
+                        floorTwinId,
+                        relationshipId,
+                        deserializeJsonString(floorContainsRoomPayload, BasicRelationship.class),
+                        BasicRelationship.class)).verifyComplete();
+                createdRelationshipIds.add(relationshipId);
             }
 
-            createRelationshipsLatch.await(MAX_WAIT_TIME_ASYNC_OPERATIONS_IN_SECONDS, TimeUnit.SECONDS);
+            // Create multiple incoming relationships to the floor. Typically a room would have relationships to multiple
+            // different floors, but for the sake of test simplicity, we'll just add multiple relationships from the same room
+            // to the same floor.
+            for (int i = 0; i < pageSize + 1; i++) {
```

Comment:
hmm, I expected this to be the `BULK_RELATIONSHIP_COUNT` instead of pageSize

Reply:
I think in this case you are only creating 5 relationships no? that's smaller than the default page size of 10

Reply:
Correct, I made a small mistake here. I was going to write this test the same way as the other pagination tests, but then I discovered we can't choose the page size here. I must have forgotten to revert this part after I made that discovery